### PR TITLE
docs: Add contributing guidelines

### PR DESCRIPTION
**Pull request by mgalka**
Add a few guidelines about filing PRs agains kernelci-core.

Signed-off-by: Michal Galka <michal.galka@collabora.com>

### CODE REVIEW

**Comment by mgalka** [comment]
This PR is gets the content of https://github.com/kernelci/kernelci-core/pull/1562 to the top level menu of the docs.

@aliceinwire @gctucker @broonie @khilman Is it ok to merge it?

**Review by nuclearcat — APPROVED** [review]
LGTM

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [inline kernelci.org/content/en/docs/contrib.md:9]: Should it be plural?

```suggestion
adding new builds, tests and device types as well as features and bugfixes for
```

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [inline kernelci.org/content/en/docs/contrib.md:12]:
```suggestion
When the PR is created, the [KernelCI staging](https://kernelci.org/docs/instances/staging)
```

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [inline kernelci.org/content/en/docs/contrib.md:15]:
```suggestion
are run on it. More detailed information about the logic behind staging runs can
```

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [inline kernelci.org/content/en/docs/contrib.md:16]:
```suggestion
be found in the [staging instance documentation](https://kernelci.org/docs/instances/staging).
```

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [inline kernelci.org/content/en/docs/contrib.md:18]:
```suggestion
There are several guidelines which can facilitate the PR review process:
```

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [inline kernelci.org/content/en/docs/contrib.md:25]: Perhaps an example of how the ideal PR comments and what sort of link to staging is expected would be useful. In particular, the links may help to understand the kind of URLs/paths expected.

Also:

```suggestion
   1. The staging instance is updated every 8h. If the PR is expected to produce build/test results, then
   check [staging dashboard](https://staging.kernelci.org) and make sure these are mentioned in the PR comment
```

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [inline kernelci.org/content/en/docs/contrib.md:33]: To avoid duplication, perhaps it is best to link to a page containing these contact details?

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [inline kernelci.org/content/en/docs/contrib.md:24]: The first sentence is already explained above -- should it be removed here?

File: kernelci.org/content/en/docs/contrib.md
Comment (gctucker) [reply kernelci.org/content/en/docs/contrib.md:16]: I think `here` links are bad, the text should be informative regardless of any hyperlinks.  That's some web design "best practice" rule I learnt a while ago...  So I'd rather leave `the staging instance documentation`.

File: kernelci.org/content/en/docs/contrib.md
Comment (gctucker) [reply kernelci.org/content/en/docs/contrib.md:33]: Namely `/docs/org/maintainers/`.

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [reply kernelci.org/content/en/docs/contrib.md:16]: Yeah, exactly (though I am not sure if you are agreeing or perhaps you read the diff the other direction).

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [reply kernelci.org/content/en/docs/contrib.md:24]: The GitHub UI makes my comment hard to understand -- I am referring here to line 24 but GitHub shows context, i.e. in particular I was talking about the "Staging instance is updated every 8h." sentence.

File: kernelci.org/content/en/docs/contrib.md
Comment (gctucker) [reply kernelci.org/content/en/docs/contrib.md:16]: Ah yes sorry, my bad :coffee:

File: kernelci.org/content/en/docs/contrib.md
Comment (aliceinwire) [inline kernelci.org/content/en/docs/contrib.md:23]: Use a permanent link for staging build/test results when possible, as current links get removed after few weeks.
maybe suggest pastebin ( like github gist )

File: kernelci.org/content/en/docs/contrib.md
Comment (aliceinwire) [inline kernelci.org/content/en/docs/contrib.md:20]: We are missing a explanation about KernelCI usage of DCO (Signed-off-by: reviewed-by: tested-by:) in comments.

File: kernelci.org/content/en/docs/contrib.md
Comment (nuclearcat) [reply kernelci.org/content/en/docs/contrib.md:23]: IMHO it is a bit hard to pastebin results from dashboard, where we have heavy ajax. Probably it is acceptable to have expiring link for new tests PR and just as confirmation that all works as designed.
There is very small chance we will need to review this test results.
But i agree that we need permanent links in case of discovered real bugs or commits.

Comment (aliceinwire) [reply kernelci.org/content/en/docs/contrib.md:23]: > where we have heavy ajax.
I'm talking mostly about logs that are on https://storage.staging.kernelci.org/ and https://storage.kernelci.org/ that are temporary

Comment (aliceinwire) [reply kernelci.org/content/en/docs/contrib.md:23]: just use something like wgetpastebin

Comment (gctucker) [reply kernelci.org/content/en/docs/contrib.md:23]: Staging builds and tests aren't permanent.

Things we can treat as permanent are mailing list archives, anything on GitHub (issues, PRs etc.) and in theory results on the production dashboard although that's more a grey area as they might disappear after a while too.

I think it's best if we can just mention some log excerpts in the commit messages directly inline, and use `Link:` for email threads like in the kernel git history.

File: kernelci.org/content/en/docs/contrib.md
Comment (mgalka) [reply kernelci.org/content/en/docs/contrib.md:20]: Is it OK to add it in as a follow-up PR?

Comment (aliceinwire) [reply kernelci.org/content/en/docs/contrib.md:20]: ok

Comment (gctucker) [reply kernelci.org/content/en/docs/contrib.md:20]: @mgalka Maybe create a GitHub issue so we don't forget about this by merging this PR?

File: kernelci.org/content/en/docs/contrib.md
Comment (mgalka) [reply kernelci.org/content/en/docs/contrib.md:15]: I'm not sure if it should be **are** I'm mentioning a _set_ here, that's why I used **is**.  Having said that I'm not a native English speaker so I might be wrong. @gctucker maybe you could help?

File: kernelci.org/content/en/docs/contrib.md
Comment (mgalka) [reply kernelci.org/content/en/docs/contrib.md:24]: I put it there as a reminder, just to make it clear that the author needs to wait at least 8h to see results on staging. On second thought, It may be too much.

File: kernelci.org/content/en/docs/contrib.md
Comment (gctucker) [reply kernelci.org/content/en/docs/contrib.md:15]: Even if the noun is singular, when it designates a plural entity it should be treated as plural e.g. "the team are on track to win the championship".  So it should be "are" imho.

Comment (ojeda) [reply kernelci.org/content/en/docs/contrib.md:15]: I think either is fine, as far as I understand it depends on what one wants to emphasize, and singular is more common in American English (https://en.wikipedia.org/wiki/American_and_British_English_grammatical_differences#Subject-verb_agreement).

File: kernelci.org/content/en/docs/contrib.md
Comment (ojeda) [reply kernelci.org/content/en/docs/contrib.md:24]: Yeah, I think either way is fine -- I didn't know if it was intentional. In some types of material, there is value on repeating key information/concepts.

One alternative is saying something shorter that reminds the reader about it, without duplicating information. For instance, something along the lines of "If build/test results are expected to be visible, then link them from staging." (which also allows you to just say "If not" in the next line, if I understand correctly)

File: kernelci.org/content/en/docs/contrib.md
Comment (mgalka) [reply kernelci.org/content/en/docs/contrib.md:25]: Actually, I looked for a PR that could be put here as an example, but all I looked into were using links to logs on stored on staging. I decided to not put it here as it was agreed that we're going to either paste log excerpts to the comments or feed data to  pastebin and paste links from there.

**Comment by mgalka** [comment]
I pushed an updated version of contributing guidelines in which I tried to addressed the comments that appeared during the review. @gctucker @ojeda @aliceinwire Could you have another look at it? If it's OK for you, I think we can land it.

**Comment by mgalka** [comment]
@gctucker @aliceinwire @ojeda 
If there are no more comments I'll merge the PR today so it can get to the production on Monday (13th of Feb).
Currently, you can see it on staging https://static.staging.kernelci.org/docs/contrib/

**Review by nuclearcat — APPROVED** [review]
great, thanks for your efforts!

File: kernelci.org/content/en/docs/contrib.md
Comment (gctucker) [reply kernelci.org/content/en/docs/contrib.md:20]: Have you created an issue for this?

Comment (gctucker) [reply kernelci.org/content/en/docs/contrib.md:20]: @mgalka See above - I haven't found any issue. I think that's the only thing to resolve before it can be merged

**Pull request merged**
